megamek/src/megamek/common/pathfinder/CachedEntityState.java

```diff
@@ -90,4 +93,25 @@ public boolean hasWorkingMisc(BigInteger flag) {
         
         return hasWorkingMisc.get(flag);
     }
+    
+    public int getTorsoJumpJets() {
+        if(torsoJumpJets == null) {
+            if(backingEntity instanceof Mech) {
+                torsoJumpJets = ((Mech) backingEntity).torsoJumpJets();
+            } else {
+                torsoJumpJets = 0;
+            }
+        }
+        
+        return torsoJumpJets;
+    }
+    
+    /**
+     * Convenience property to determine if the backing entity is amphibious.
+     */
+    public boolean isAmphibious() {
+        return hasWorkingMisc(MiscType.F_FULLY_AMPHIBIOUS) || 
+                hasWorkingMisc(MiscType.F_AMPHIBIOUS) ||
+                hasWorkingMisc(MiscType.F_FULLY_AMPHIBIOUS);
```

Comment:
Should one of these be F_LIMITED_AMPHIBIOUS?

Reply:
Yes.